cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowContext.java

```diff
@@ -51,4 +52,10 @@
    * @return a {@link WorkflowToken}
    */
   WorkflowToken getToken();
+
+  /**
+   * @return the {@link WorkflowActionContext} for custom actions
+   */
+  @Nullable
+  WorkflowActionContext getWorkflowActionContext();
```

Comment:
Only custom action have access to `WorkflowContext`, so can have the WorkflowContext extends from DatasetContext directly. No need to have extra layer of indirection.